server/src/utils/getConfig.ts

```diff
@@ -12,36 +12,69 @@ import {
+      process.env[VV_COMMUNICATION_SERVICES_CONNECTION_STRING] ?? defaultConfig.communicationServicesConnectionString,
+    microsoftBookingsUrl: process.env[VV_MICROSOFT_BOOKINGS_URL_ENV_NAME] ?? defaultConfig.microsoftBookingsUrl,
     chatEnabled:
       typeof process.env[VV_CHAT_ENABLED_ENV_NAME] === 'string'
         ? process.env[VV_CHAT_ENABLED_ENV_NAME]?.toLowerCase() === 'true'
-        : DefaultConfig.chatEnabled,
+        : defaultConfig.chatEnabled,
     screenShareEnabled:
       typeof process.env[VV_SCREENSHARE_ENABLED_ENV_NAME] === 'string'
         ? process.env[VV_SCREENSHARE_ENABLED_ENV_NAME]?.toLowerCase() === 'true'
-        : DefaultConfig.screenShareEnabled,
-    companyName: process.env[VV_COMPANY_NAME_ENV_NAME] ?? DefaultConfig.companyName,
-    colorPalette: process.env[VV_COLOR_PALETTE_ENV_NAME] ?? DefaultConfig.colorPalette,
-    waitingTitle: process.env[VV_WAITING_TITLE_ENV_NAME] ?? DefaultConfig.waitingTitle,
-    waitingSubtitle: process.env[VV_WAITING_SUBTITLE_ENV_NAME] ?? DefaultConfig.waitingSubtitle,
-    logoUrl: process.env[VV_LOGO_URL_ENV_NAME] ?? DefaultConfig.logoUrl
-  };
+        : defaultConfig.screenShareEnabled,
+    companyName: process.env[VV_COMPANY_NAME_ENV_NAME] ?? defaultConfig.companyName,
+    colorPalette: process.env[VV_COLOR_PALETTE_ENV_NAME] ?? defaultConfig.colorPalette,
+    waitingTitle: process.env[VV_WAITING_TITLE_ENV_NAME] ?? defaultConfig.waitingTitle,
+    waitingSubtitle: process.env[VV_WAITING_SUBTITLE_ENV_NAME] ?? defaultConfig.waitingSubtitle,
+    logoUrl: process.env[VV_LOGO_URL_ENV_NAME] ?? defaultConfig.logoUrl
+  } as ServerConfigModel;
+
+  //Check environment variables for values for postCall and set them if present
+  //else check defaultConfig.json for postCall config values and use them
+  let postCallType: PostCallSurveyType;
+  let postCallSurveyUrl;
+  if (
+    typeof process.env[VV_POSTCALL_SURVEY_TYPE_ENV_NAME] !== undefined &&
+    process.env[VV_POSTCALL_SURVEY_OPTIONS_SURVEYURL_ENV_NAME] !== undefined &&
+    (process.env[VV_POSTCALL_SURVEY_TYPE_ENV_NAME] === 'msforms' ||
+      process.env[VV_POSTCALL_SURVEY_TYPE_ENV_NAME] === 'thirdparty')
+  ) {
+    postCallType = <PostCallSurveyType>process.env[VV_POSTCALL_SURVEY_TYPE_ENV_NAME]?.toLowerCase();
+    postCallSurveyUrl =
+      process.env[VV_POSTCALL_SURVEY_OPTIONS_SURVEYURL_ENV_NAME] !== undefined
+        ? process.env[VV_POSTCALL_SURVEY_OPTIONS_SURVEYURL_ENV_NAME]
+        : '';
+
+    config.postCall = {
+      survey: { type: postCallType, options: { surveyUrl: postCallSurveyUrl } }
+    };
+  } else if (defaultConfig.postCall?.survey?.type) {
+    postCallType = <PostCallSurveyType>defaultConfig.postCall?.survey?.type;
+    postCallSurveyUrl = defaultConfig.postCall?.survey?.options?.surveyUrl;
+    config.postCall = {
+      survey: { type: postCallType, options: { surveyUrl: postCallSurveyUrl } }
+    };
+  }
```

Comment:
We can simplify/clarify the logic here:
- Set the initial values to the default values
- Try and get each post call environment variable. If found, overwrite the default.

We should also move this logic to a new function for separation of concerns. PoC: https://github.com/Azure-Samples/communication-services-virtual-visits-js/commit/0bbe84a60b65b258419d084bdbcb92499ea039b6#diff-169359eb8c23e178028aa80922c7a185a667bc1732bf59178a66eb2ab78ffdc3R60

In addition, we should write the appropriate unit tests, including a test for handling a value that can't be serialized to a `PostCallSurveyType`

Reply:
This change has been made